network/default.go

```diff
@@ -1058,28 +1058,37 @@ func (n *network) processNetChan(listener tunnel.Listener) {
 					// lookup best routes for the services in the just received route
 					q := []router.QueryOption{
 						router.QueryService(route.Service),
-						router.QueryStrategy(router.AdvertiseBest),
+						router.QueryStrategy(n.router.Options().Advertise),
 					}
-					bestRoutes, err := n.options.Router.Table().Query(q...)
+
+					routes, err := n.options.Router.Table().Query(q...)
 					if err != nil {
 						log.Debugf("Network node %s failed listing best routes for %s: %v", n.id, route.Service, err)
 						continue
 					}
 
-					// we found no route for the given service
-					// create new route we have just received
-					if len(bestRoutes) == 0 {
+					// we found no routes for the given service
+					// create the new route we have just received
+					if len(routes) == 0 {
 						// add routes to the routing table
 						if err := n.router.Table().Create(route); err != nil && err != router.ErrDuplicateRoute {
```

Comment:
If you're creating the route here should you just continue?

Reply:
That's the change I was thinking of when I messaged you in slack hah. Let me think...